util/plugins/eu.esdihumboldt.util.test/src/eu/esdihumboldt/util/geometry/interpolation/CircleInterpolationTest.java

```diff
@@ -0,0 +1,145 @@
+
+/**
+ * Test for Circle Type
+ * 
+ * @author Arun
+ */
+@RunWith(Parameterized.class)
+public class CircleInterpolationTest {
+
+	private final int testIndex;
+	private final Coordinate[] arcCoordinates;
+	@SuppressWarnings("rawtypes")
+	private final Class generatedGeometryType;
+	private final boolean skipTest;
+	private static final double e = 0.1;
+
+	private static final boolean SKIP_TEST = false;
+
+	private static final boolean DRAW_IMAGE = false;
+
+	private static final boolean DEFAULT_KEEP_ORIGINAL = true;
+
+	/**
+	 * Constructor for parameterized test
+	 * 
+	 * @param testIndex Index of test parameters
+	 * @param coordinates input arc coordinates
+	 * @param geometry type of output geometry
+	 * @param skipTest if wants to skip test
+	 */
+	@SuppressWarnings("rawtypes")
+	public CircleInterpolationTest(int testIndex, Coordinate[] coordinates, Class geometry,
+			boolean skipTest) {
+		this.testIndex = testIndex;
+		this.arcCoordinates = coordinates;
+		this.generatedGeometryType = geometry;
+		this.skipTest = skipTest;
+	}
+
+	/**
+	 * Passing circle geometries
+	 * 
+	 * @return Collection of arc coordinates and type of generated geometry
+	 */
+	@SuppressWarnings("rawtypes")
+	@Parameters
+	public static Collection addCoordiantes() {
+		return Arrays.asList(new Object[][] { //
+				{ 0, new Coordinate[] { new Coordinate(569884.075, 5936816.054),
+						new Coordinate(569883.230, 5936814.518),
+						new Coordinate(569884.919, 5936814.518) }, //
+						LineString.class, SKIP_TEST }, //
+				{ 1, new Coordinate[] { new Coordinate(568420.259, 5936349.171),
+						new Coordinate(568419.414, 5936347.635),
+						new Coordinate(568421.103, 5936347.635) }, //
+						LineString.class, SKIP_TEST }, //
+				{ 2, new Coordinate[] { new Coordinate(8, 8), new Coordinate(12, 16),
+						new Coordinate(16, 8) }, //
+						LineString.class, SKIP_TEST }, //
+				{ 3, new Coordinate[] { new Coordinate(240, 280), new Coordinate(210, 150),
+						new Coordinate(300, 100) }, //
+						LineString.class, SKIP_TEST }, //
+				{ 4, new Coordinate[] { new Coordinate(8, 8), new Coordinate(12, 6.5),
+						new Coordinate(16, 8) }, //
+						LineString.class, SKIP_TEST }, //
+				{ 5, new Coordinate[] { new Coordinate(3, 10.5), new Coordinate(4, 7.75),
+						new Coordinate(8, 8) }, //
+						LineString.class, SKIP_TEST }, //
+				{ 6, new Coordinate[] { new Coordinate(569675.954, 5937944.689),
+						new Coordinate(569675.109, 5937943.153),
+						new Coordinate(569676.798, 5937943.153) }, //
+						LineString.class, SKIP_TEST }, //
+				{ 7, new Coordinate[] { new Coordinate(0.01, 3.2), new Coordinate(3.33, 3.33),
+						new Coordinate(0.01, -3.2) }, //
+						LineString.class, SKIP_TEST } //
+		});
+	}
+
+//0.01,3.2 3.33,3.33 0.01,-3.2
+	/**
+	 * test algorithm
+	 */
+	@Test
+	public void testInterpolation() {
+		System.out.println("-- Test-" + testIndex + " begin --");
+		if (skipTest) {
```

Comment:
What is the purpose of this condition?

The danger I see here that looking at the test results you will never know that a test was actually skipped, because it will be listed as passed test.

Why should tests be skipped?
If there is a use case for that it might be wiser to use JUnit `assume*` methods (see [here](http://stackoverflow.com/questions/1689242/conditionally-ignoring-tests-in-junit-4) for an example).